### PR TITLE
Add Jenkinsfile for sandbox deployment

### PR DESCRIPTION
**Pull request by banderous**
### Change description ###
Add a parameterized Jenkinsfile based on producer repository.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_parameterized [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    name: secretName,
+    version: '',
+    envVariable: envVar
+  ]
+}
+
+properties([
+  parameters([
+    string(name: 'PRODUCT_NAME', defaultValue: 'send-letter', description: ''),
+    string(name: 'APP', defaultValue: 'send-letter-service', description: ''),
```
Comment (lgonczar) [inline Jenkinsfile_parameterized]: I think this should just be `service`.  The pipeline appends it to the product name.

File: Jenkinsfile_parameterized [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!groovy
+@Library("Infrastructure") _
+
+List<LinkedHashMap<String, Object>> secrets = [
+  secret('smoke-test-s2s-name', 'SMOKE_TEST_S2S_NAME'),
```
Comment (lgonczar) [inline Jenkinsfile_parameterized]: These keys are currently not set by terraform, which will cause the job to fail. You would need to specify them in main.tf, just like it's done in the producer.

Comment (banderous) [reply Jenkinsfile_parameterized]: I've updated the main terraform script.

File: infrastructure/main.tf [hunk 3]
```diff
@@ -92,4 +97,16 @@ resource "azurerm_key_vault_secret" "smoke-test-s2s-url" {
   value     = "${var.s2s_url}"
   vault_uri = "${module.key-vault.key_vault_uri}"
 }
+
+resource "azurerm_key_vault_secret" "smoke-test-s2s-name" {
+  name      = "smoke-test-s2s-name"
+  value     = "send_letter_tests"
```
Comment (rkondratowicz) [inline infrastructure/main.tf]: Is this service meant to be used in both functional and smoke tests?
If so, it would be worth renaming the secret.

**Pull request merged**